middleware.go

```diff
@@ -385,63 +375,50 @@ func handleMultipart(c *Client, r *Request) (err error) {
 	for k, v := range r.FormData {
 		for _, iv := range v {
 			if strings.HasPrefix(k, "@") { // file
-				err = addFile(w, k[1:], iv)
-				if err != nil {
-					return
+				if err := addFile(w, k[1:], iv); err != nil {
+					return err
 				}
 			} else { // form value
-				if err = w.WriteField(k, iv); err != nil {
+				if err := w.WriteField(k, iv); err != nil {
 					return err
 				}
 			}
 		}
 	}
 
 	// #21 - adding io.Reader support
-	if len(r.multipartFiles) > 0 {
-		for _, f := range r.multipartFiles {
-			err = addFileReader(w, f)
-			if err != nil {
-				return
-			}
+	for _, f := range r.multipartFiles {
+		if err := addFileReader(w, f); err != nil {
+			return err
 		}
 	}
 
 	// GitHub #130 adding multipart field support with content type
-	if len(r.multipartFields) > 0 {
-		for _, mf := range r.multipartFields {
-			if err = addMultipartFormField(w, mf); err != nil {
-				return
-			}
+	for _, mf := range r.multipartFields {
+		if err := addMultipartFormField(w, mf); err != nil {
+			return err
 		}
 	}
 
 	r.Header.Set(hdrContentTypeKey, w.FormDataContentType())
-	err = w.Close()
-
-	return
+	return w.Close()
 }
 
 func handleFormData(c *Client, r *Request) {
 	formData := url.Values{}
-
-	for k, v := range c.FormData {
-		for _, iv := range v {
-			formData.Add(k, iv)
-		}
-	}
-
 	for k, v := range r.FormData {
-		// remove form data field from client level by key
-		// since overrides happens for that key in the request
-		formData.Del(k)
+		formData[k] = v[:]
```

Comment:
@SVilgelm Any reason you did not take the approach you did for the request header processing here to prevent the new variable `formData := url.Values{}`?

Reply:
Almost same, just didn't updater the request's form data in place, probably I should. Will update later today

Reply:
Thanks @SVilgelm

Reply:
done